### PR TITLE
Add full vehicle entity parsing

### PR DESCRIPTION
**Pull request by cedarbaum**
Based on discussion in #3, add full support of GTFS realtime vehicle entities:

- Update `parseVehicle` to handle all vehicle properties and add test
- Update `vehicle` hash function and update existing test
- Print full vehicle entities in the CLI
  - Can be tested by using a snapshot of the MTA vehicle GTFS feed here: http://bt.mta.info/wiki/Developers/GTFSRt

### CODE REVIEW

File: realtime.go [hunk 1]
```diff
@@ -99,16 +99,37 @@ type StopTimeEvent struct {
...
+
+	CurrentStopSequence *uint32
+
+	StopID *string
+
+	CurrentStatus *CurrentStatus
+
+	Timestamp *time.Time
+
+	CongestionLevel *CongestionLevel
```
Comment (jamespfennell) [inline realtime.go:127]: Would it be possible to make this not a pointer, and if the field is unset then set it to `UNKNOWN_CONGESTION_LEVEL`?

File: realtime.go [hunk 2]
```diff
@@ -99,16 +99,37 @@ type StopTimeEvent struct {
...
+
+	OccupancyStatus *OccupancyStatus
```
Comment (jamespfennell) [inline realtime.go:129]: and also make this non-nil, with a default of `NO_DATA_AVAILABLE`? If you think it would be good.

File: realtime.go [hunk 3]
```diff
@@ -361,7 +390,7 @@ func parseVehicle(vehiclePosition *gtfsrt.VehiclePosition, opts *ParseRealtimeOp
 		ID:                parseTripDescriptor(vehiclePosition.Trip, opts),
 		IsEntityInMessage: false,
 	}
-	return trip, vehicle, false
+	return trip, vehicle, true
```
Comment (jamespfennell) [inline realtime.go:393]: It looks like the last field is always true now so we could get rid of the last return value, and at the call site set `ok = true`

File: realtime.go [hunk 4]
```diff
@@ -99,16 +99,37 @@ type StopTimeEvent struct {
...
+type CurrentStatus = gtfsrt.VehiclePosition_VehicleStopStatus
+type CongestionLevel = gtfsrt.VehiclePosition_CongestionLevel
+type OccupancyStatus = gtfsrt.VehiclePosition_OccupancyStatus
+
 type Vehicle struct {
 	ID *VehicleID
 
 	Trip *Trip
 
+	Position *Position
```
Comment (jamespfennell) [inline realtime.go:117]: Is there any benefit right now to creating a non-proto-generate `Position` struct and mapping accordingly? Maybe not?

File: realtime_test.go [hunk 5]
```diff
@@ -170,6 +171,106 @@ func TestAlert(t *testing.T) {
...
+	return &s
+}
+
+func ptrCongestionLevel(s gtfsrt.VehiclePosition_CongestionLevel) *gtfsrt.VehiclePosition_CongestionLevel {
+	return &s
+}
+
+func ptrOccupancyStatus(s gtfsrt.VehiclePosition_OccupancyStatus) *gtfsrt.VehiclePosition_OccupancyStatus {
+	return &s
+}
```
Comment (jamespfennell) [inline realtime_test.go:276]: If it makes your life easier, now that Go has generics all of these functions could be replaced by a single one:

```
func ptr[T any](t T) *T {
  return &t
}
```

File: hash.go [hunk 6]
```diff
@@ -74,10 +70,27 @@ func (h *hasher) trip(t *Trip) {
 func (h *hasher) vehicle(v *Vehicle) {
```
Comment (jamespfennell) [inline hash.go:70]: In addition to these changes, we likely need a public function to support hashing a vehicle? Something like:
```
func (v *Vehicle) Hash(h hash.Hash) {
	s := hasher{h: h}
	s.vehicle(v)
	s.flush()
}
```

**Review by jamespfennell — CHANGES_REQUESTED** [review]
Thanks for taking the time to do this! Just a few small comments, nothing major I hope, but LGTM overall.

File: hash.go (hunk 6 above)
Comment (cedarbaum) [reply hash.go:70]: I believe this function already exists (`hash.go`, line 23).

File: realtime_test.go (hunk 5 above)
Comment (cedarbaum) [reply realtime_test.go]: Great suggestion, replaced all functions with a generic version.

File: realtime.go (hunk 2 above)
Comment (cedarbaum) [reply realtime.go:129]: I didn't see a good default (something that clearly means "value not present") for `OccupancyStatus` or `CurrentStatus` enums unfortunately, unless I am missing something. Or was the suggestion here to create new enums?

File: realtime.go (hunk 4 above)
Comment (cedarbaum) [reply realtime.go:117]: Good question, I couldn't think of a specific reason to do it for now so I just followed the existing pattern of type aliasing some existing gtfsrt proto types (e.g., how this was done for some enum types). That being said, I am happy to create a new struct to further decouple this library from the proto types if you think that would be cleaner.

File: hash.go (hunk 6 above)
Comment (jamespfennell) [reply hash.go:70]: Woops, yes!

File: realtime.go (hunk 4 above)
Comment (jamespfennell) [reply realtime.go:117]: I think for the moment we can submit as-is, and if wanted later change it.

File: realtime.go (hunk 2 above)
Comment (jamespfennell) [reply realtime.go:129]: My (hacky!) suggestion was to piggyback on the existing enum value `NO_DATA_AVAILABLE`. But maybe let's again punt it. In the future we may add a non-proto enum, and then we can have a specific value that represents "the feed did not set this field".

**Review by jamespfennell — APPROVED** [review]
Sorry for the late response, was visiting family in California! :)

**Comment by jamespfennell** [comment]
Thank you for doing this!

**Pull request merged**